packages/components/src/border-control/types.ts

```diff
@@ -0,0 +1,160 @@
+export type Border = {
+	color?: string;
+	style?: string;
+	width?: string;
+};
+
+export type Color = {
+	name: string;
+	color: string;
+};
+
+export type BorderControlProps = {
+	/**
+	 * An array of color definitions. This may also be a multi-dimensional array
+	 * where colors are organized by multiple origins.
+	 */
+	colors?: Color[];
+	/**
+	 * This toggles the ability to choose custom colors.
+	 */
+	disableCustomColors?: boolean;
+	/**
+	 * This controls whether the alpha channel will be offered when selecting
+	 * custom colors.
+	 */
+	enableAlpha?: boolean;
+	/**
+	 * Provides control over whether the label will only be visible to screen
+	 * readers.
+	 */
+	hideLabelFromVision?: boolean;
+	/**
+	 * This flags the `BorderControl` to render with a more compact appearance.
+	 * It restricts the width of the control and prevents it from expanding to
+	 * take up additional space.
+	 */
+	isSmall?: boolean;
+	/**
+	 * If provided, a label will be generated using this as the content.
+	 */
+	label?: string;
+	/**
+	 * A callback function invoked when the border value is changed via an
+	 * interaction that selects or clears, border color, style, or width.
+	 */
+	onChange: ( value: Border | undefined ) => void;
+	/**
+	 * If opted into, sanitizing the border means that if no width or color have
+	 * been selected, the border style is also cleared and `undefined`
+	 * is returned as the new border value.
+	 */
+	shouldSanitizeBorder?: boolean;
+	/**
+	 * This controls whether to include border style options within the
+	 * `BorderDropdown` sub-component.
+	 */
+	showStyle?: boolean;
+	/**
+	 * An object representing a border or `undefined`. Used to set the current
+	 * border configuration for this component.
+	 */
+	value?: Border;
+	/**
+	 * Controls the visual width of the `BorderControl`.
+	 */
+	width?: string;
```

Comment:
Why do we need a `width` prop, when we can style the component with CSS if needed?

(Also, in case we were to keep this prop, have you considered typing it using the `CSSProperties` type?)

Reply:
This was for convenience. It also helps automatically update the flex styling so it takes affect properly. Yes, as a consumer we could add a classname to the root component and then dig down to target the appropriate control. 

I could hardcode these styles for the `BorderBoxControl` but thought it nicer devx to offer it for other consumers as well.

Reply:
I see! Let's leave this prop for now, I'll see if I can think of a good alternative in the meantime